src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/KeyValuePairListEncoding.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug;
+
+ using System.Collections.Generic;
+    using System.Linq;
+
+    internal sealed class KeyValuePairListEncoding
+    {
+        public IEnumerable<(string Name, string Value)> Parse(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                yield break;
+            }
+
+            foreach (var entry in ReadEntries(input))
+            {
+                var (entryKey, entryValue) = SplitEntry(entry);
+                var decodedEntryKey = Decode(entryKey);
+                var decodedEntryValue = Decode(entryValue);
+
+                if (!string.IsNullOrEmpty(decodedEntryKey))
+                {
+                    yield return (decodedEntryKey, decodedEntryValue);
+                }
+            }
+
+            static IEnumerable<string> ReadEntries(string rawText)
+            {
+                bool escaped = false;
+                int entryStart = 0;
+                for (int i = 0; i < rawText.Length; i++)
+                {
+                    if (rawText[i] == ',' && !escaped)
+                    {
+                        yield return rawText.Substring(entryStart, i - entryStart);
+                        entryStart = i + 1;
+                        escaped = false;
+                    }
+                    else if (rawText[i] == '/')
+                    {
+                        escaped = !escaped;
+                    }
+                    else
+                    {
+                        escaped = false;
+                    }
+                }
+
+                yield return rawText.Substring(entryStart);
+            }
+
+            static (string EncodedKey, string EncodedValue) SplitEntry(string entry)
+            {
+                bool escaped = false;
+                for (int i = 0; i < entry.Length; i++)
+                {
+                    if (entry[i] == '=' && !escaped)
+                    {
+                        return (entry.Substring(0, i), entry.Substring(i + 1));
+                    }
+                    else if (entry[i] == '/')
+                    {
+                        escaped = !escaped;
+                    }
+                    else
+                    {
+                        escaped = false;
+                    }
+                }
+
+                return (string.Empty, string.Empty);
+            }
```

Comment:
I spent some time looking at these since the logic is written as (in English):
> If `[character]` is proceeded by `/`, ignore it while splitting by `[character]`.

There's a very non-obtuse way to do that, `Regex.Split`. It is like `String.Split`, but does it based on a regular expression instead of just a character. Here's an example for the first function, where you want to split on `,`, but not on `/,`.

```C#
// See https://aka.ms/new-console-template for more information
using System.Text.RegularExpressions;

var string1 = "fdsafdsa,fdsafdsa,fdasfdsa/,dsafdsa/,dasfdsa,fdsa,fdsa,";
var string2 = "fgdg;g/g/fds/gfd;,,;;,;,gfds;.g;fd.s;,;fd.s;///fd;s,gf;ds.g;fd/,/gfd,/,d/sg,f/ds";

var regex = new Regex(@"(?<!\/),");
Console.WriteLine("1:");
regex.Split(string1).ToList().ForEach(Console.WriteLine);
Console.WriteLine("2:");
regex.Split(string2).ToList().ForEach(Console.WriteLine);
```

The expression used is `(?<!\/),` which says, "Match `,` that isn't preceded by `/`". The result is:
![image](https://user-images.githubusercontent.com/17788297/178645177-667ffd70-196a-4863-92ba-e80cef921f69.png)

This works the same as your code. You can do the same for `=` in the second function. Here's some simpler ways I wrote the function before using `Regex.Split`.

This is the same as yours but removes the boolean because it is completely redundant.
```C#
static IEnumerable<string> ReadEntries(string rawText)
{
    int entryStart = 0;
    for (int i = 0; i < rawText.Length; i++)
    {
        // The character prior to ',' cannot be '/'.
        if (rawText[i] == ',' && !(i > 0 && rawText[i - 1] == '/'))
        {
            yield return rawText.Substring(entryStart, i - entryStart);
            entryStart = i + 1;
        }
    }

    yield return rawText.Substring(entryStart);
}
```

Second one here splits them based on `,` but then combines those that end in `/`. It is not efficient but allowed me to realize I simply wanted a conditional `String.Split` (which is what `Regex.Split` is).
```C#
static IEnumerable<string> ReadEntries2(string rawText)
{
    var entries = rawText.Split(',');
    var current = string.Empty;
    foreach(var entry in entries)
    {
        // If the entry ends with a "/", we want to combine it with subsequent entries.
        if (entry.EndsWith("/"))
        {
            current += entry;
            continue;
        }

        yield return current + entry;
        current = string.Empty;
    }
}
```